crates/oxc_transformer/src/typescript/annotations.rs

```diff
@@ -325,4 +327,236 @@ impl<'a> TypeScriptAnnotations<'a> {
             }
         }
     }
+
+    /// ```TypeScript
+    /// enum Foo {
+    ///   X
+    /// }
+    /// ```
+    /// ```JavaScript
+    /// var Foo = ((Foo) => {
+    ///   const X = 0; Foo[Foo["X"] = X] = "X";
+    ///   return Foo;
+    /// })(Foo || {});
+    /// ```
+    fn transform_ts_enum(
```

Comment:
Can we isolate this in a new file?

Reply:
Yeah I was gonna do this in an enums file.

Reply:
Yes, i moved this to `enum.rs`